trie/hasher.go

```diff
@@ -125,7 +125,7 @@ func (h *hasher) hashChildren(original node, db *Database) (node, node, error) {
 	case *shortNode:
```

Comment:
This took me a second too, but since `node` is a polymorphic type, when we have a var of the `node` interface type, we need to figure out which actual implementation it is to handle it properly. This does that determination and assignment to `n` in one go. Also see: https://tour.golang.org/methods/16